modules/mobile/app/models/mobile/v0/service_graph.rb

```diff
@@ -31,9 +31,10 @@ def affected_services(windows)
 
           while queue.size != 0
             s = queue.shift
-            if s && s.name != name && s.leaf?
+            dss = downstream_windows[s.name]
+            if s && s.name != name && s.leaf? && (!dss || window.start_time < dss&.start_time)
```

Comment:
don't love that all these conditions are all together in one if statement but rubocop didn't like it any other way.

Reply:
I think this might be slightly clearer with `dss.present?` instead of `!dss`. Dealing with the inversion within a long series of conditions is harder for the brain to wrap around.

Reply:
sounds good. updated to use `nil?`